src/views/waypoint/home/index.tsx

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-case-declarations */
+/* eslint-disable react/no-array-index-key */
+import React, { ReactElement } from 'react'
+import DocsLayout from 'layouts/docs'
+import GetStarted, { GetStartedProps } from './components/get-started'
+import Cards, { CardProps } from './components/cards'
+import Heading, { HeadingProps } from './components/heading'
+
+function WaypointLanding({
+  CONTENT,
+}: {
+  CONTENT: { heading: string; subheading: string; blocks: Block[] }
+}): ReactElement {
+  return (
+    <div>
+      <h1>{CONTENT.heading}</h1>
+      <p>{CONTENT.subheading}</p>
+      <LandingBlocks blocks={CONTENT.blocks} />
+    </div>
+  )
+}
+
+type Block =
+  | ({ type: 'heading' } & HeadingProps)
+  | ({ type: 'get_started' } & GetStartedProps)
+  | ({ type: 'cards' } & CardProps)
+
+function LandingBlocks({ blocks }: { blocks: Block[] }) {
+  return (
+    <>
+      {blocks.map((block, idx) => {
+        const { type } = block
+        if (type === 'heading') {
+          const { heading, __heading_slug, level } = block
+          return <Heading key={idx} {...{ __heading_slug, heading, level }} />
+        } else if (type === 'get_started') {
+          const { heading, iconSvg, text, link } = block
+          return <GetStarted key={idx} {...{ iconSvg, heading, text, link }} />
+        } else if (type === 'cards') {
+          const { columns, cards } = block
+          return <Cards key={idx} {...{ columns, cards }} />
+        }
```

Comment:
I would say this or a `switch` statement is the correct way to go with a discriminated union type like you've got!